strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/test/java/org/carlspring/strongbox/providers/layout/RawLayoutProviderTest.java

```diff
@@ -68,81 +60,47 @@
-
-        return repositories;
-    }
-
-    @BeforeEach
-    public void setUp()
-            throws Exception
-    {
-        Configuration configuration = configurationManagementService.getConfiguration();
-
-        if (configuration.getStorage(STORAGE) == null)
-        {
-            createStorage(STORAGE);
-        }
-
-        if (configuration.getStorage(STORAGE).getRepository(REPOSITORY) == null)
-        {
-            createRepository(STORAGE, REPOSITORY);
-        }
-    }
-
-    @AfterEach
-    public void removeRepositories()
-            throws IOException, JAXBException
-    {
-        removeRepositories(getRepositoriesToClean());
-    }
-
     @Test
-    public void testNullArtifactCoordinates()
+    void testNullArtifactCoordinates()
     {
         NullArtifactCoordinates coordinates = new NullArtifactCoordinates("foo/bar/blah.bz2");
 
-        System.out.println("coordinates.toPath(): " + coordinates.toPath());
+        logger.info("coordinates.toPath(): {}", coordinates.toPath());
     }
 
+    @ExtendWith({RepositoryManagementTestExecutionListener.class,
+                 ArtifactManagementTestExecutionListener.class })
     @Test
-    public void testDeployAndResolveArtifact()
+    void testDeployAndResolveArtifact(@TestRepository(layout = RawLayoutProvider.ALIAS,
+                                                      storageId = STORAGE,
+                                                      repositoryId = REPOSITORY)
+                                     Repository repository,
+                                     @TestArtifact(resource = "foo/bar.zip",
+                                                   generator = NullArtifactGenerator.class)
+                                     Path artifactPath)
             throws Exception
     {
         String path = "foo/bar.zip";
 
-        RepositoryPath repositoryPath = repositoryPathResolver.resolve(STORAGE, REPOSITORY, path);
+        RepositoryPath artifactRepositoryPath = repositoryPathResolver.resolve(repository, path);
 
         // Deploy the artifact
-        artifactManagementService.validateAndStore(repositoryPath, createZipFile());
+        artifactManagementService.validateAndStore(artifactRepositoryPath, createZipFile());
 
-        
-        File artifactFile = new File(propertiesBooter.getVaultDirectory() + "/storages/" + STORAGE + "/" + REPOSITORY + "/" + path);
-        assertTrue(artifactFile.exists(), "Failed to deploy artifact!");
-        assertTrue(artifactFile.length() > 0, "Failed to deploy artifact!");
+        assertTrue(Files.exists(artifactRepositoryPath), "Failed to deploy artifact!");
+        assertTrue(Files.size(artifactRepositoryPath) > 0, "Failed to deploy artifact!");
 
         // Attempt to re-deploy the artifact
         try
         {
-            artifactManagementService.validateAndStore(repositoryPath, createZipFile());
+            artifactManagementService.validateAndStore(artifactRepositoryPath, createZipFile());
```

Comment:
@ptirador I've just noticed that we don't need `createZipFile()` here as well, you should use `Files.newInputStream(artifactPath)` instead,